dev/main.go

```diff
@@ -33,12 +36,18 @@ func New(
 	}
 
 	return &DaggerDev{
-		Source:    source,
+		Src:       source,
 		Version:   versionInfo,
 		DockerCfg: dockerCfg,
 	}, nil
 }
 
+// Enable module auto-codegen when retrieving the dagger source code
+func (dev *DaggerDev) WithModCodegen() *DaggerDev {
```

Comment:
I think this is alright, but because with can be applied to the entire dev module, it's possible for the user to enable it, which could potentially interfere with the operation of the module. Ideally it would be private to avoid this.

I still don't quite understand why `Source`/`GeneratedSource` isn't alright here, but, I suppose this is fine as is.